src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -90,19 +98,19 @@ public void enqueue(KafkaMessage message) {
 		}
 	}
 
-	@Override
 	public void start() {
 		this.running = true;
 	}
 
-	@Override
-	public void stop() {
+	public void stop(long stopTimeout) {
+		shutdownLatch = new CountDownLatch(1);
 		this.running = false;
```

Comment:
We can add an option to drain the internal before stopping but I don't think that should be the default.